x-pack/plugins/transform/public/app/sections/create_transform/components/aggregation_list/popover_form.tsx

```diff
@@ -325,6 +367,23 @@ export const PopoverForm: React.FC<Props> = ({ defaultData, otherAggNames, onCha
           />
         </EuiFormRow>
       )}
+      {agg === PIVOT_SUPPORTED_AGGS.TERMS && (
+        <EuiFormRow
+          label={i18n.translate('xpack.transform.agg.popoverForm.sizeLabel', {
+            defaultMessage: 'Size',
+          })}
+          error={
+            !validSize && [
+              i18n.translate('xpack.transform.groupBy.popoverForm.invalidSizeErrorMessage', {
+                defaultMessage: 'Enter a valid positive number',
+              }),
+            ]
+          }
+          isInvalid={!validSize}
+        >
+          <EuiFieldText defaultValue={sizeText} onChange={(e) => updateSize(e.target.value)} />
```

Comment:
size is a `number` type, shall we use https://elastic.github.io/eui/#/forms/form-controls#number-field control instead?

Reply:
I remember we intentionally went with the text field in other cases too because we had problems in the past with custom validations. So for consistency I went with the text based one here again. See the input for `'Maximum page search size'` for example.

Reply:
Might be worth another look at `EuiFieldNumber` as currently it silently accepts fractional values (rounding down to nearest integer):

![image](https://user-images.githubusercontent.com/7405507/151021594-f842a151-fde6-419c-971b-4c8008e03368.png)

Reply:
Thanks for the additional feedback, I switched to `EuiFieldNumber` in 4878e18. Looking into this, I noticed the form validation approach I copied over from the percentiles variant was a bit flawed. I fixed that in the same commit. For both percentiles and size it's now validating the input string and keeping that as separate state. Previously, similar like the screenshot above with a float and not error, for percentiles it also wouldn't show an error but just fall back to the default value when applied.